documentation/Language/Locale.md

```diff
@@ -0,0 +1,47 @@
+# Locale
+
+This feature of the language allows specifing localization tokens and values for languages indentified by [ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) codes.
+
+```mint
+locale en {
+  ui: {
+    buttons: {
+      ok: "OK"
+    }
+  }
+}
```

Comment:
It's a good start! I like the explicit locale strings definitions 👍 
How do you express the variables/interpolation within strings?
Also, this doesn't seem to support explicit pluralization rules.

Reply:
It's not just strings, the values can be any type / expression so it's easy to do interpolation / pluralization with functions:

```mint
locale en {
  greet: (name : String) { "Hello #{name}!" },
  article: (howMany : Number) {
    if (howMany == 0 && howMany == 1) {
      "Article"
    } else {
      "Articles"
    }
  }
}
```
The type signature of the values must match between locales.

Reply:
I see. It's an option, alright, although it seems like a PITA to repeat a lot of boilerplate code in order to achieve these two functionalities.

Reply:
What boilerplate code are you referring to? For each language?

Reply:
I appreciate that the values are just expressions. I've not done much localisation but I assume it can get quite complicated depending on the target language.

Perhaps for some common tasks there could be some methods in the core library?

```mint
locale en {
  article: Localisation.plural("No Articles", "1 Article", (count : String) { "#{count} Articles" })
}
```

Reply:
@gdotdesign I meant pluralization, @jansul beat me to it :)

Reply:
There is no reason why we can't have functions like in the `Locale` module but I'm not sure that all languages have pluralization like that :thinking: which a quick google search confirms https://stackoverflow.com/questions/6166064/i18n-pluralization :sweat_smile: 

So maybe we will need to have separate modules for each languages functions `Locale.EN.pluralize()`, `Locale.RU.pluralize()`, etc...

Reply:
Separate module per language seems to make sense, especially given that Russian example!

My only other concern for this feature would be the file size if an application includes a lot of languages, but that would have to wait until something like code-splitting was added to Mint.

Reply:
I'd try to generalize it within a single module, but load the language rules dynamically. Having `zero`, `one`, `few`, `many` and `other` keys (with fallbacks if undefined, e.g. `few || other`) should handle most of the cases. Kinda like how Rails is [handling](https://guides.rubyonrails.org/i18n.html#pluralization) that.

Reply:
I think it's out of the scope of the current PR :slightly_smiling_face: so I'm resolving this in order to get it merge.